contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpExtractor.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.http;
+
+import io.opencensus.common.ExperimentalApi;
+import javax.annotation.Nullable;
+
+/**
+ * An adaptor to extract information from request and response.
+ *
+ * <p>This class provides no-op implementations by default.
+ *
+ * <p>Please refer to this <a
+ * href="https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md">document</a>
+ * for more information about the HTTP attributes recorded in Open Census.
+ *
+ * @param <Q> the HTTP request entity.
+ * @param <P> the HTTP response entity.
+ * @since 0.17
+ */
+@ExperimentalApi
+public abstract class HttpExtractor<Q, P> {
```

Comment:
We should also have a way to get the route:
https://github.com/census-instrumentation/opencensus-specs/blob/master/stats/HTTP.md#tags-1

Reply:
I also add getUrl() as per https://github.com/census-instrumentation/opencensus-specs/pull/183/files.